### PR TITLE
GitHub Action 

### PR DESCRIPTION
**Pull request by dougb**
This is a very basic GitHub Action. Checking in to share progress.
To test locally on Mac, install act via `brew tap nektos/tap && brew install nektos/tap/act`

Issues:
- one of the IO tests fails. 
- After running locally with [act](https://github.com/nektos/act) a few times, docker runs out of disk space somewhere.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0")
 addSbtPlugin("com.typesafe.sbt" % "sbt-git" % "1.0.0")
 addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.9.0")
+addSbtPlugin("io.get-coursier" % "sbt-coursier" % "1.0.3")
```
Comment (colindean) [inline project/plugins.sbt]: 🎉

**Comment by dougb** [comment]
@phpisciuneri Just pointed me to this [issue](/nektos/act/issues#60) that sounds exactly like my problem.

**Comment by phpisciuneri** [comment]
@doug https://stackoverflow.com/questions/55110729/how-do-i-cache-steps-in-github-actions seems like there is limited discussion here on the caching issue as well

**Comment by phpisciuneri** [comment]
@dougb I get the same test failure that you were getting.  I haven't replicated the "no space on device" error yet.  I have rerun about 15 times using `act -r`.

The test failure is due to the random generation of the test string: https://github.com/target/data-validator/blob/master/src/test/scala/com/target/data_validator/IOSpec.scala#L51

I compared char by char and they all fail except for the newline.  The problem appears to be related to this: https://stackoverflow.com/questions/31345150/scala-random-nextstringint-returning-question-marks

We are generating random chars, majority of which are not renderable, so that when they get written to file they are written as a question mark.  Then when we read from the file, we think we are reading the actual question mark char and the comparison fails.  I converted to character codes and dumped to stdout before and after reading from the file.  Here is example output:

```
Vector(43990, 4043, 19049, 14592, 17813, 46538, 41715, 40739, 44647, 43865, 38539, 47651, 9312, 54252, 7005, 10206, 55153, 30568, 30112, 11735, 15967, 22821, 46411, 16298, 20426, 36172, 49340, 28069, 28553, 7589, 23811, 26834, 24822, 10486, 18583, 4312, 1305, 47842, 13357, 32603, 35659, 21999, 48546, 27503, 22604, 27487, 12894, 29327, 22322, 25716, 53789, 39543, 51139, 5040, 47524, 28762, 55057, 39459, 35980, 28559, 5538, 39193, 41703, 8273, 12351, 42756, 2374, 20674, 18951, 43443, 48851, 983, 40916, 27255, 49246, 6396, 46289, 5680, 4946, 39909, 23765, 42885, 31795, 18500, 42189, 13771, 52819, 18175, 13638, 29746, 18724, 52957, 42192, 36779, 15948, 54238, 54316, 25262, 14290, 52650, 12124, 640, 22284, 44664, 20830, 9409, 4628, 25147, 46714, 48040, 25735, 41714, 19892, 44535, 18019, 2680, 46338, 5420, 18383, 10545, 42385, 14069, 45252, 16902, 33249, 35848, 35317, 30693, 10)
Vector(63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 63, 10)
[info] - file:/// should be able to write *** FAILED ***
[info]   "[????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????]
[info]   " was not equal to "[????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????]
[info]   " (IOSpec.scala:56)
```

I think we can just swap out `nextString` for `nextPrintableChar`.  This works for me:

```
val data = List.fill(128)(nextPrintableChar).mkString + IO.NEW_LINE
```

**Comment by phpisciuneri** [comment]
@dougb I think we might need a workflow for `push` (for additional commits to a PR) and maybe `release` events as well.  It doesn't seem that the `on` takes lists.

**Pull request merged**